src/plugins/data/public/search/session/session_state.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import uuid from 'uuid';
+import { Observable } from 'rxjs';
+import { distinctUntilChanged, map, shareReplay } from 'rxjs/operators';
+import { createStateContainer, StateContainer } from '../../../../kibana_utils/public';
+
+/**
+ * Possible state that current session can be in
+ *
+ * @public
+ */
+export enum SessionState {
```

Comment:
Should these types be aligned with [these](src/plugins/data/common/search/session/status.ts)?

Reply:
I agree that those seem to overlap, but those serve different purposes and represent different things:

* `src/plugins/data/public/search/session/session_state.ts` represents a client side state of the current session service.
* `https://github.com/elastic/kibana/blob/master/src/plugins/data/common/search/session/status.ts` is basically ~ a subset of those states. This represent a status of existing background session SO

I don't think we should try to unite those, but probably it should be named better. Maybe `SessionServiceState`